version: "3.3"

services:

  element-1:
    container_name: element-${SUB_DOMAIN_1}
    image: vectorim/element-web:v1.11.31
    volumes:
      - ./${SUB_DOMAIN_1}/element-config.json:/app/config.json
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.element-${SUB_DOMAIN_1}.rule=Host(`element.${SUB_DOMAIN_1}.${DOMAIN}`)"
      - "traefik.http.routers.element-${SUB_DOMAIN_1}.entrypoints=websecure"
      - "traefik.http.services.element-${SUB_DOMAIN_1}.loadbalancer.server.port=80"
      - "traefik.http.routers.element-${SUB_DOMAIN_1}.tls=true"
      - "traefik.http.routers.element-${SUB_DOMAIN_1}.tls.certresolver=letsencrypt"

  element-2:
    container_name: element-${SUB_DOMAIN_2}
    image: vectorim/element-web:v1.11.31
    volumes:
      - ./${SUB_DOMAIN_2}/element-config.json:/app/config.json
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.element-${SUB_DOMAIN_2}.rule=Host(`element.${SUB_DOMAIN_2}.${DOMAIN}`)"
      - "traefik.http.routers.element-${SUB_DOMAIN_2}.entrypoints=websecure"
      - "traefik.http.services.element-${SUB_DOMAIN_2}.loadbalancer.server.port=80"
      - "traefik.http.routers.element-${SUB_DOMAIN_2}.tls=true"
      - "traefik.http.routers.element-${SUB_DOMAIN_2}.tls.certresolver=letsencrypt"

  synapse-1:
    container_name: synapse-${SUB_DOMAIN_1}
    image: matrixdotorg/synapse:${SYNAPSE_VERSION}
    volumes:
      - ./${SUB_DOMAIN_1}/mx-data:/data
      - ./${SUB_DOMAIN_1}/mx-conf:/mx-conf/
      - ./broadcast_module/:/usr/local/lib/python3.11/site-packages/broadcast_module
      - ./synapse/handlers/receipts.py:/usr/local/lib/python3.11/site-packages/synapse/handlers/receipts.py
    environment:
      - SYNAPSE_CONFIG_PATH=/mx-conf/homeserver.yaml
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_1}.rule=Host(`${SUB_DOMAIN_1}.${DOMAIN}`)"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_1}.entrypoints=websecure"
      - "traefik.http.services.synapse-${SUB_DOMAIN_1}.loadbalancer.server.port=8008"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_1}.tls=true"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_1}.tls.certresolver=letsencrypt"

  synapse-2:
    container_name: synapse-${SUB_DOMAIN_2}
    image: matrixdotorg/synapse:${SYNAPSE_VERSION}
    volumes:
      - ./${SUB_DOMAIN_2}/mx-data:/data
      - ./${SUB_DOMAIN_2}/mx-conf:/mx-conf/
      - ./broadcast_module/:/usr/local/lib/python3.11/site-packages/broadcast_module
      - ./synapse/handlers/receipts.py:/usr/local/lib/python3.11/site-packages/synapse/handlers/receipts.py
    environment:
      - SYNAPSE_CONFIG_PATH=/mx-conf/homeserver.yaml
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_2}.rule=Host(`${SUB_DOMAIN_2}.${DOMAIN}`)"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_2}.entrypoints=websecure"
      - "traefik.http.services.synapse-${SUB_DOMAIN_2}.loadbalancer.server.port=8008"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_2}.tls=true"
      - "traefik.http.routers.synapse-${SUB_DOMAIN_2}.tls.certresolver=letsencrypt"

networks:
  default:
    external: true
    name: ${TRAEFIK_NETWORK}
